### PR TITLE
antlr4: 4.8.x -> 4.11.x

### PR DESCRIPTION
**Pull request by mweinelt**
- Update the default antlr4 version away from a two year old release.
- Remove the multiple versions present in the python package set, we don't support that use case

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/baserow/default.nix [hunk 1]
```diff
@@ -2,11 +2,20 @@
...
-  baserow_premium = with python3.pkgs; ( buildPythonPackage rec {
+  python = python3.override {
+    packageOverrides = self: super: {
+      antlr4-python3-runtime = super.antlr4-python3-runtime.override {
+        antlr4 = antlr4_9;
+      };
+    };
+  };
+
+  baserow_premium = with python.pkgs; ( buildPythonApplication rec {
```
Comment (dotlambda) [inline pkgs/servers/baserow/default.nix:18]: Why not add this to the overrides and use `buildPythonPackage`?

Comment (mweinelt) [reply pkgs/servers/baserow/default.nix:18]: I'm not sure what baserow_premium is, it's only used in checkInputs. I'm leaving that to @onny.

Comment (dotlambda) [reply pkgs/servers/baserow/default.nix:18]: It used `buildPythonPackage` before after all.

Comment (onny) [reply pkgs/servers/baserow/default.nix:18]: `baserow_premium` is a dependency of `baserow`. Not sure how to handle this properly, I'm free for any recommendations :)

**Comment by dotlambda** [comment]
Let remove the JAR file in `postPatch`.

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 207851` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>antlr</li>
    <li>antlr4_8</li>
    <li>azure-cli</li>
    <li>baserow</li>
    <li>luaformatter</li>
    <li>python310Packages.antlr4-python3-runtime</li>
    <li>python310Packages.explorerscript</li>
    <li>python310Packages.hydra-core</li>
    <li>python310Packages.omegaconf</li>
    <li>python310Packages.skytemple-dtef</li>
    <li>python310Packages.skytemple-files</li>
    <li>python310Packages.skytemple-ssb-debugger</li>
    <li>python39Packages.antlr4-python3-runtime</li>
    <li>python39Packages.explorerscript</li>
    <li>python39Packages.hydra-core</li>
    <li>python39Packages.omegaconf</li>
    <li>python39Packages.skytemple-dtef</li>
    <li>python39Packages.skytemple-files</li>
    <li>python39Packages.skytemple-ssb-debugger</li>
    <li>skytemple</li>
  </ul>
</details>

File: pkgs/servers/baserow/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/servers/baserow/default.nix]: Reverted to `buildPythonPackage` and integrated into the packageOverrides.

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 207851` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>antlr</li>
    <li>antlr4_8</li>
    <li>azure-cli</li>
    <li>baserow</li>
    <li>luaformatter</li>
    <li>python310Packages.antlr4-python3-runtime</li>
    <li>python310Packages.explorerscript</li>
    <li>python310Packages.hydra-core</li>
    <li>python310Packages.omegaconf</li>
    <li>python310Packages.skytemple-dtef</li>
    <li>python310Packages.skytemple-files</li>
    <li>python310Packages.skytemple-ssb-debugger</li>
    <li>python39Packages.antlr4-python3-runtime</li>
    <li>python39Packages.explorerscript</li>
    <li>python39Packages.hydra-core</li>
    <li>python39Packages.omegaconf</li>
    <li>python39Packages.skytemple-dtef</li>
    <li>python39Packages.skytemple-files</li>
    <li>python39Packages.skytemple-ssb-debugger</li>
    <li>skytemple</li>
  </ul>
</details>

File: pkgs/development/python-modules/explorerscript/default.nix [hunk 2]
```diff
@@ -26,8 +27,33 @@ buildPythonPackage rec {
...
+  ] ++ passthru.optional-dependencies.pygments;
+
+  disabledTestPaths = [
+    # test fixtures are incompatible with more recent antlr4 versions
+    "tests/compiler/macros_imports_test.py"
+  ];
+
+  pythonImportsCheck = [
+    "explorerscript"
+  ];
```
Comment (xfix) [inline pkgs/development/python-modules/explorerscript/default.nix:56]: Please mark the package as broken instead or regenerate contents of `explorerscript/antlr` directory with a newer version of ANTLR as a build step.

**Review by sugar700 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/explorerscript/default.nix (hunk 2 above)
Comment (mweinelt) [reply pkgs/development/python-modules/explorerscript/default.nix:56]: Feel free to propose a way to do that. This is the error I run into when running the tests.

```pytb
____________ ERROR collecting tests/compiler/macros_imports_test.py ____________
tests/compiler/macros_imports_test.py:27: in <module>
    from explorerscript.ssb_converting.ssb_compiler import ExplorerScriptSsbCompiler
explorerscript/ssb_converting/ssb_compiler.py:29: in <module>
    from explorerscript.explorerscript_reader import ExplorerScriptReader
explorerscript/explorerscript_reader.py:25: in <module>
    from explorerscript.antlr.ExplorerScriptLexer import ExplorerScriptLexer
explorerscript/antlr/ExplorerScriptLexer.py:371: in <module>
    class ExplorerScriptLexer(Lexer):
explorerscript/antlr/ExplorerScriptLexer.py:373: in ExplorerScriptLexer
    atn = ATNDeserializer().deserialize(serializedATN())
/nix/store/hjqv9m18z9233bblqlw9lrdng39f830c-python3.10-antlr4-python3-runtime-4.11.1/lib/python3.10/site-packages/antlr4/atn/ATNDeserializer.py:28: in deserialize
    self.checkVersion()
/nix/store/hjqv9m18z9233bblqlw9lrdng39f830c-python3.10-antlr4-python3-runtime-4.11.1/lib/python3.10/site-packages/antlr4/atn/ATNDeserializer.py:50: in checkVersion
    raise Exception("Could not deserialize ATN with version " + str(version) + " (expected " + str(SERIALIZED_VERSION) + ").")
E   Exception: Could not deserialize ATN with version  (expected 4).
```

Comment (dotlambda) [reply pkgs/development/python-modules/explorerscript/default.nix:56]: Maybe @theCapypara can tell us how to do so?

Comment (xfix) [reply pkgs/development/python-modules/explorerscript/default.nix:56]: Running `antlr -Dlanguage=Python3 explorerscript/antlr/*.g4` with a newer version of antlr should work.

Comment (mweinelt) [reply pkgs/development/python-modules/explorerscript/default.nix:56]: Yeah, that seems to have done something. But it looks liker newer antlr4-python-runtime versions return a list, were previous ones returned a string. Maybe it's best to take this upstream now.

```
============================= test session starts ==============================
platform linux -- Python 3.10.9, pytest-7.1.3, pluggy-1.0.0
rootdir: /build/source
collected 12 items                                                             

tests/compiler/macros_imports_test.py FFFFFFFFFFFF                       [100%]

=================================== FAILURES ===================================
______________________ MacrosImportsTestCase.test_complex ______________________

self = <tests.compiler.macros_imports_test.MacrosImportsTestCase testMethod=test_complex>

    def test_complex(self):
>       self.assert_can_compile("test_complex")

tests/compiler/macros_imports_test.py:60: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/compiler/macros_imports_test.py:75: in assert_can_compile
    self.compile(fixture_name)
tests/compiler/macros_imports_test.py:87: in compile
    compiler.compile(exps_source, fixture_path)
explorerscript/ssb_converting/ssb_compiler.py:133: in compile
    for subfile_path in self._resolve_imported_file(os.path.dirname(file_name)):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <explorerscript.ssb_converting.ssb_compiler.ExplorerScriptSsbCompiler object at 0x7ffff62a7dc0>
dir_name = '/build/source/tests/fixtures/compiler/macros_imports_test/test_complex'

    def _resolve_imported_file(self, dir_name):
        """
        Returns the full paths to all imports specified in self.imports.
        If any file can not be found, raises an SsbCompilerError.
        """
        fs = []
        for import_file in self.imports:
>           if import_file.startswith('.') or import_file.startswith('/'):
E           AttributeError: 'list' object has no attribute 'startswith'

explorerscript/ssb_converting/ssb_compiler.py:204: AttributeError
___________________ MacrosImportsTestCase.test_err_notfound ____________________

self = <tests.compiler.macros_imports_test.MacrosImportsTestCase testMethod=test_err_notfound>

    def test_err_notfound(self):
>       self.assert_can_not_compile("test_err_notfound", SsbCompilerError)

tests/compiler/macros_imports_test.py:63: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/compiler/macros_imports_test.py:78: in assert_can_not_compile
    self.assertRaises(error, self.compile, fixture_name)
tests/compiler/macros_imports_test.py:87: in compile
    compiler.compile(exps_source, fixture_path)
explorerscript/ssb_converting/ssb_compiler.py:133: in compile
    for subfile_path in self._resolve_imported_file(os.path.dirname(file_name)):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def _resolve_imported_file(self, dir_name):
        """
        Returns the full paths to all imports specified in self.imports.
        If any file can not be found, raises an SsbCompilerError.
        """
        fs = []
        for import_file in self.imports:
>           if import_file.startswith('.') or import_file.startswith('/'):
E           AttributeError: 'list' object has no attribute 'startswith'

explorerscript/ssb_converting/ssb_compiler.py:204: AttributeError
____________________ MacrosImportsTestCase.test_err_params _____________________

self = <tests.compiler.macros_imports_test.MacrosImportsTestCase testMethod=test_err_params>

    def test_err_params(self):
>       self.assert_can_not_compile("test_err_params", ValueError)

tests/compiler/macros_imports_test.py:66: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/compiler/macros_imports_test.py:78: in assert_can_not_compile
    self.assertRaises(error, self.compile, fixture_name)
tests/compiler/macros_imports_test.py:87: in compile
    compiler.compile(exps_source, fixture_path)
explorerscript/ssb_converting/ssb_compiler.py:133: in compile
    for subfile_path in self._resolve_imported_file(os.path.dirname(file_name)):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def _resolve_imported_file(self, dir_name):
        """
        Returns the full paths to all imports specified in self.imports.
        If any file can not be found, raises an SsbCompilerError.
        """
        fs = []
        for import_file in self.imports:
>           if import_file.startswith('.') or import_file.startswith('/'):
E           AttributeError: 'list' object has no attribute 'startswith'

explorerscript/ssb_converting/ssb_compiler.py:204: AttributeError
__________________ MacrosImportsTestCase.test_err_recursion1 ___________________

self = <tests.compiler.macros_imports_test.MacrosImportsTestCase testMethod=test_err_recursion1>

    def test_err_recursion1(self):
>       self.assert_can_not_compile("test_err_recursion1", SsbCompilerError)

tests/compiler/macros_imports_test.py:69: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/compiler/macros_imports_test.py:78: in assert_can_not_compile
    self.assertRaises(error, self.compile, fixture_name)
tests/compiler/macros_imports_test.py:87: in compile
    compiler.compile(exps_source, fixture_path)
explorerscript/ssb_converting/ssb_compiler.py:133: in compile
    for subfile_path in self._resolve_imported_file(os.path.dirname(file_name)):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def _resolve_imported_file(self, dir_name):
        """
        Returns the full paths to all imports specified in self.imports.
        If any file can not be found, raises an SsbCompilerError.
        """
        fs = []
        for import_file in self.imports:
>           if import_file.startswith('.') or import_file.startswith('/'):
E           AttributeError: 'list' object has no attribute 'startswith'

explorerscript/ssb_converting/ssb_compiler.py:204: AttributeError
__________________ MacrosImportsTestCase.test_err_recursion2 ___________________

self = <tests.compiler.macros_imports_test.MacrosImportsTestCase testMethod=test_err_recursion2>

    def test_err_recursion2(self):
>       self.assert_can_not_compile("test_err_recursion2", SsbCompilerError)

tests/compiler/macros_imports_test.py:72: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/compiler/macros_imports_test.py:78: in assert_can_not_compile
    self.assertRaises(error, self.compile, fixture_name)
tests/compiler/macros_imports_test.py:87: in compile
    compiler.compile(exps_source, fixture_path)
explorerscript/ssb_converting/ssb_compiler.py:133: in compile
    for subfile_path in self._resolve_imported_file(os.path.dirname(file_name)):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def _resolve_imported_file(self, dir_name):
        """
        Returns the full paths to all imports specified in self.imports.
        If any file can not be found, raises an SsbCompilerError.
        """
        fs = []
        for import_file in self.imports:
>           if import_file.startswith('.') or import_file.startswith('/'):
E           AttributeError: 'list' object has no attribute 'startswith'

explorerscript/ssb_converting/ssb_compiler.py:204: AttributeError
______________________ MacrosImportsTestCase.test_nested _______________________

self = <tests.compiler.macros_imports_test.MacrosImportsTestCase testMethod=test_nested>

    def test_nested(self):
>       self.assert_can_compile("test_nested")

tests/compiler/macros_imports_test.py:48: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/compiler/macros_imports_test.py:75: in assert_can_compile
    self.compile(fixture_name)
tests/compiler/macros_imports_test.py:87: in compile
    compiler.compile(exps_source, fixture_path)
explorerscript/ssb_converting/ssb_compiler.py:133: in compile
    for subfile_path in self._resolve_imported_file(os.path.dirname(file_name)):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <explorerscript.ssb_converting.ssb_compiler.ExplorerScriptSsbCompiler object at 0x7ffff62c65f0>
dir_name = '/build/source/tests/fixtures/compiler/macros_imports_test/test_nested'

    def _resolve_imported_file(self, dir_name):
        """
        Returns the full paths to all imports specified in self.imports.
        If any file can not be found, raises an SsbCompilerError.
        """
        fs = []
        for import_file in self.imports:
>           if import_file.startswith('.') or import_file.startswith('/'):
E           AttributeError: 'list' object has no attribute 'startswith'

explorerscript/ssb_converting/ssb_compiler.py:204: AttributeError
___________________ MacrosImportsTestCase.test_nested_chain ____________________

self = <tests.compiler.macros_imports_test.MacrosImportsTestCase testMethod=test_nested_chain>

    def test_nested_chain(self):
>       self.assert_can_compile("test_nested_chain")

tests/compiler/macros_imports_test.py:51: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/compiler/macros_imports_test.py:75: in assert_can_compile
    self.compile(fixture_name)
tests/compiler/macros_imports_test.py:87: in compile
    compiler.compile(exps_source, fixture_path)
explorerscript/ssb_converting/ssb_compiler.py:133: in compile
    for subfile_path in self._resolve_imported_file(os.path.dirname(file_name)):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <explorerscript.ssb_converting.ssb_compiler.ExplorerScriptSsbCompiler object at 0x7ffff6467550>
dir_name = '/build/source/tests/fixtures/compiler/macros_imports_test/test_nested_chain'

    def _resolve_imported_file(self, dir_name):
        """
        Returns the full paths to all imports specified in self.imports.
        If any file can not be found, raises an SsbCompilerError.
        """
        fs = []
        for import_file in self.imports:
>           if import_file.startswith('.') or import_file.startswith('/'):
E           AttributeError: 'list' object has no attribute 'startswith'

explorerscript/ssb_converting/ssb_compiler.py:204: AttributeError
____________________ MacrosImportsTestCase.test_other_file _____________________

self = <tests.compiler.macros_imports_test.MacrosImportsTestCase testMethod=test_other_file>

    def test_other_file(self):
>       self.assert_can_compile("test_other_file")

tests/compiler/macros_imports_test.py:54: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/compiler/macros_imports_test.py:75: in assert_can_compile
    self.compile(fixture_name)
tests/compiler/macros_imports_test.py:87: in compile
    compiler.compile(exps_source, fixture_path)
explorerscript/ssb_converting/ssb_compiler.py:133: in compile
    for subfile_path in self._resolve_imported_file(os.path.dirname(file_name)):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <explorerscript.ssb_converting.ssb_compiler.ExplorerScriptSsbCompiler object at 0x7ffff676ad70>
dir_name = '/build/source/tests/fixtures/compiler/macros_imports_test/test_other_file'

    def _resolve_imported_file(self, dir_name):
        """
        Returns the full paths to all imports specified in self.imports.
        If any file can not be found, raises an SsbCompilerError.
        """
        fs = []
        for import_file in self.imports:
>           if import_file.startswith('.') or import_file.startswith('/'):
E           AttributeError: 'list' object has no attribute 'startswith'

explorerscript/ssb_converting/ssb_compiler.py:204: AttributeError
_____________________ MacrosImportsTestCase.test_other_two _____________________

self = <tests.compiler.macros_imports_test.MacrosImportsTestCase testMethod=test_other_two>

    def test_other_two(self):
>       self.assert_can_compile("test_other_two")

tests/compiler/macros_imports_test.py:57: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/compiler/macros_imports_test.py:75: in assert_can_compile
    self.compile(fixture_name)
tests/compiler/macros_imports_test.py:87: in compile
    compiler.compile(exps_source, fixture_path)
explorerscript/ssb_converting/ssb_compiler.py:133: in compile
    for subfile_path in self._resolve_imported_file(os.path.dirname(file_name)):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <explorerscript.ssb_converting.ssb_compiler.ExplorerScriptSsbCompiler object at 0x7ffff6a20b80>
dir_name = '/build/source/tests/fixtures/compiler/macros_imports_test/test_other_two'

    def _resolve_imported_file(self, dir_name):
        """
        Returns the full paths to all imports specified in self.imports.
        If any file can not be found, raises an SsbCompilerError.
        """
        fs = []
        for import_file in self.imports:
>           if import_file.startswith('.') or import_file.startswith('/'):
E           AttributeError: 'list' object has no attribute 'startswith'

explorerscript/ssb_converting/ssb_compiler.py:204: AttributeError
______________________ MacrosImportsTestCase.test_simple1 ______________________

self = <tests.compiler.macros_imports_test.MacrosImportsTestCase testMethod=test_simple1>

    def test_simple1(self):
>       self.assert_can_compile("test_simple1")

tests/compiler/macros_imports_test.py:39: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/compiler/macros_imports_test.py:75: in assert_can_compile
    self.compile(fixture_name)
tests/compiler/macros_imports_test.py:87: in compile
    compiler.compile(exps_source, fixture_path)
explorerscript/ssb_converting/ssb_compiler.py:133: in compile
    for subfile_path in self._resolve_imported_file(os.path.dirname(file_name)):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <explorerscript.ssb_converting.ssb_compiler.ExplorerScriptSsbCompiler object at 0x7ffff62c6320>
dir_name = '/build/source/tests/fixtures/compiler/macros_imports_test/test_simple1'

    def _resolve_imported_file(self, dir_name):
        """
        Returns the full paths to all imports specified in self.imports.
        If any file can not be found, raises an SsbCompilerError.
        """
        fs = []
        for import_file in self.imports:
>           if import_file.startswith('.') or import_file.startswith('/'):
E           AttributeError: 'list' object has no attribute 'startswith'

explorerscript/ssb_converting/ssb_compiler.py:204: AttributeError
______________________ MacrosImportsTestCase.test_simple2 ______________________

self = <tests.compiler.macros_imports_test.MacrosImportsTestCase testMethod=test_simple2>

    def test_simple2(self):
>       self.assert_can_compile("test_simple2")

tests/compiler/macros_imports_test.py:42: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/compiler/macros_imports_test.py:75: in assert_can_compile
    self.compile(fixture_name)
tests/compiler/macros_imports_test.py:87: in compile
    compiler.compile(exps_source, fixture_path)
explorerscript/ssb_converting/ssb_compiler.py:133: in compile
    for subfile_path in self._resolve_imported_file(os.path.dirname(file_name)):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <explorerscript.ssb_converting.ssb_compiler.ExplorerScriptSsbCompiler object at 0x7ffff62a57e0>
dir_name = '/build/source/tests/fixtures/compiler/macros_imports_test/test_simple2'

    def _resolve_imported_file(self, dir_name):
        """
        Returns the full paths to all imports specified in self.imports.
        If any file can not be found, raises an SsbCompilerError.
        """
        fs = []
        for import_file in self.imports:
>           if import_file.startswith('.') or import_file.startswith('/'):
E           AttributeError: 'list' object has no attribute 'startswith'

explorerscript/ssb_converting/ssb_compiler.py:204: AttributeError
___________________ MacrosImportsTestCase.test_simple3_chain ___________________

self = <tests.compiler.macros_imports_test.MacrosImportsTestCase testMethod=test_simple3_chain>

    def test_simple3_chain(self):
>       self.assert_can_compile("test_simple3_chain")

tests/compiler/macros_imports_test.py:45: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/compiler/macros_imports_test.py:75: in assert_can_compile
    self.compile(fixture_name)
tests/compiler/macros_imports_test.py:87: in compile
    compiler.compile(exps_source, fixture_path)
explorerscript/ssb_converting/ssb_compiler.py:133: in compile
    for subfile_path in self._resolve_imported_file(os.path.dirname(file_name)):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <explorerscript.ssb_converting.ssb_compiler.ExplorerScriptSsbCompiler object at 0x7ffff649aef0>
dir_name = '/build/source/tests/fixtures/compiler/macros_imports_test/test_simple3_chain'

    def _resolve_imported_file(self, dir_name):
        """
        Returns the full paths to all imports specified in self.imports.
        If any file can not be found, raises an SsbCompilerError.
        """
        fs = []
        for import_file in self.imports:
>           if import_file.startswith('.') or import_file.startswith('/'):
E           AttributeError: 'list' object has no attribute 'startswith'

explorerscript/ssb_converting/ssb_compiler.py:204: AttributeError

Comment (xfix) [reply pkgs/development/python-modules/explorerscript/default.nix:56]: Yeah, it's fine, mark the package as broken for now (it essentially is), I will look into this later.

Comment (xfix) [reply pkgs/development/python-modules/explorerscript/default.nix:56]: Okay, it's necessary to pass `-visitor` flag: `antlr -Dlanguage=Python3 -visitor explorerscript/antlr/{ExplorerScript,SsbScript}.g4`

Comment (mweinelt) [reply pkgs/development/python-modules/explorerscript/default.nix]: Applied, thanks!

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 207851` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>22 packages built:</summary>
  <ul>
    <li>antlr</li>
    <li>antlr4_8</li>
    <li>azure-cli</li>
    <li>baserow</li>
    <li>luaformatter</li>
    <li>python310Packages.antlr4-python3-runtime</li>
    <li>python310Packages.explorerscript</li>
    <li>python310Packages.hassil</li>
    <li>python310Packages.hydra-core</li>
    <li>python310Packages.omegaconf</li>
    <li>python310Packages.skytemple-dtef</li>
    <li>python310Packages.skytemple-files</li>
    <li>python310Packages.skytemple-ssb-debugger</li>
    <li>python39Packages.antlr4-python3-runtime</li>
    <li>python39Packages.explorerscript</li>
    <li>python39Packages.hassil</li>
    <li>python39Packages.hydra-core</li>
    <li>python39Packages.omegaconf</li>
    <li>python39Packages.skytemple-dtef</li>
    <li>python39Packages.skytemple-files</li>
    <li>python39Packages.skytemple-ssb-debugger</li>
    <li>skytemple</li>
  </ul>
</details>

Result of `nixpkgs-review pr 207851` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>22 packages built:</summary>
  <ul>
    <li>antlr</li>
    <li>antlr4_8</li>
    <li>azure-cli</li>
    <li>baserow</li>
    <li>luaformatter</li>
    <li>python310Packages.antlr4-python3-runtime</li>
    <li>python310Packages.explorerscript</li>
    <li>python310Packages.hassil</li>
    <li>python310Packages.hydra-core</li>
    <li>python310Packages.omegaconf</li>
    <li>python310Packages.skytemple-dtef</li>
    <li>python310Packages.skytemple-files</li>
    <li>python310Packages.skytemple-ssb-debugger</li>
    <li>python39Packages.antlr4-python3-runtime</li>
    <li>python39Packages.explorerscript</li>
    <li>python39Packages.hassil</li>
    <li>python39Packages.hydra-core</li>
    <li>python39Packages.omegaconf</li>
    <li>python39Packages.skytemple-dtef</li>
    <li>python39Packages.skytemple-files</li>
    <li>python39Packages.skytemple-ssb-debugger</li>
    <li>skytemple</li>
  </ul>
</details>

**Pull request merged**